web/styles/rendered_markdown.css

```diff
@@ -23,15 +23,31 @@
         margin-left: 0;
     }
 
-    /* Ensure ordered lists are nicely centered in 1-line messages */
+    /* Ensure ordered lists are nicely centered in 1-line messages for upto 2
+       digit markers, else the markers take up as much space as needed without
+       being cut off at the beginning */
     & ol {
-        margin: 2px 0 5px 20px;
+        margin: 2px 0 5px;
+
+        & li {
+            display: block;
+        }
+
+        & li::before {
+            content: counter(list-item) ". ";
+            text-align: right;
+            counter-increment: list-item;
+            display: inline-block;
+            min-width: 20px;
+            margin-right: 3px;
+        }
     }
 
-    /* Swap the left and right margins of ordered list for Right-To-Left languages */
-    &.rtl ol {
-        margin-right: 8px;
-        margin-left: 0;
+    /* Invert direction of text alignment and margin of markers in ordered lists
+       for proper centering in Right-To-Left languages */
+    &.rtl ol li::before {
+        text-align: left;
+        margin-left: 3px;
```

Comment:
Do we need to unset the `margin-right: 3px` that you had above for this case?

Reply:
> Also could this CSS be added as `.rlt &`, inside the above `::before` block? I'm not sure but that might feel a bit more readable.

I tried, but that does not work syntactically

Reply:
OK